homeassistant/components/iotty/switch.py

```diff
@@ -0,0 +1,165 @@
+"""Implement a iotty Light Switch Device."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from iottycloud.device import Device
+from iottycloud.lightswitch import LightSwitch
+from iottycloud.verbs import LS_DEVICE_TYPE_UID
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .api import IottyProxy
+from .const import DOMAIN, KNOWN_DEVICES
+from .coordinator import IottyDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Activate the iotty LightSwitch component."""
+    _LOGGER.debug("Setup SWITCH entry id is %s", config_entry.entry_id)
+
+    hass_data = hass.data[DOMAIN]
+
+    coordinator: IottyDataUpdateCoordinator = hass_data[config_entry.entry_id]
+
+    entities = [
+        IottyLightSwitch(
+            coordinator=coordinator, iotty_cloud=coordinator.iotty, iotty_device=d
+        )
+        for d in coordinator.data.devices
+        if d.device_type == LS_DEVICE_TYPE_UID
+    ]
+    _LOGGER.debug("Found %d LightSwitches", len(entities))
+
+    async_add_entities(entities)
+
+    @callback
+    def async_update_data():
+        """Handle updated data from the API endpoint."""
+        if not coordinator.last_update_success:
+            return
+
+        devices = coordinator.data.devices
+        entities = []
+        known_devices: set = hass.data[DOMAIN][KNOWN_DEVICES]
+
+        # Add entities for devices which we've not yet seen
+        for device in devices:
+            if device in known_devices:
+                continue
+
+            entities.extend(
+                [
+                    IottyLightSwitch(
+                        coordinator=coordinator,
+                        iotty_cloud=coordinator.iotty,
+                        iotty_device=d,
+                    )
+                    for d in coordinator.data.devices
+                    if d.device_type == LS_DEVICE_TYPE_UID
+                    and d.device_id == device.device_id
```

Comment:
What's up with the looping here? The outer loop already loops over `coordinator.data.devices`, why do we loop over it *again*?

Reply:
My bad, it's fixed now